### PR TITLE
feat(37): add login page, route, and cubit

### PR DESCRIPTION
**Pull request by stephenrichter**
This PR adds the UI scaffolding for the LoginPage, route, cubit, and placeholder for AuthRepository.

Simulator Screenshot - iPhone 15 Pro Max - 2024-05-22 at 09 48 07

### CODE REVIEW

File: lib/auth/login/widgets/login_form.dart [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+class LoginForm extends StatefulWidget {
+  const LoginForm({
+    super.key,
+  });
+
+  @override
+  State<LoginForm> createState() => _LoginFormState();
+}
+
+class _LoginFormState extends State<LoginForm> {
```
Comment (ncuillery) [inline lib/auth/login/widgets/login_form.dart]: From what I see, this widget has no reason to be stateful (no local variable and no lifecycle method overriden)

File: lib/auth/login/cubits/login_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }) async {
+    try {
+      // start loading
+      emit(state.copyWith(
+        status: Status.pending,
+        email: email.trim(),
+      ));
+
+      // save data if remember me is checked
+      if (state.rememberMe) {
```
Comment (ncuillery) [inline lib/auth/login/cubits/login_cubit.dart:39]: "Remember me" means "stay connected" right? Not keeping the prefilling the email field 🙂

File: lib/auth/login/widgets/login_form.dart [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                const SizedBox(height: FlexSizes.inputFieldSpacing),
+
+                // Remember me and forgot password
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    // Remember me
+                    Row(
+                      children: [
+                        Checkbox(
```
Comment (ncuillery) [inline lib/auth/login/widgets/login_form.dart:78]: How about using `FormBuilderCheckbox` here?

File: lib/cart/widgets/cart_content.dart [hunk 4]
```diff
@@ -25,7 +27,7 @@ class CartContent extends StatelessWidget {
           OrderSummary(cart: cart),
           const SizedBox(height: 8),
           ElevatedButton(
-            onPressed: () {},
+            onPressed: () => context.router.push(const LoginRoute()),
```
Comment (ncuillery) [inline lib/cart/widgets/cart_content.dart]: pushNamed?

File: lib/shared/utils/image_strings.dart [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class FlexImages {
+  // TODO Replace with the media permalink corresponding to this image
```
Comment (ncuillery) [inline lib/shared/utils/image_strings.dart:2]: We can find it in our backoffice now, right?

**Review by ncuillery — APPROVED** [review]

File: lib/auth/login/cubits/login_cubit.dart (hunk 2 above)
Comment (stephenrichter) [reply lib/auth/login/cubits/login_cubit.dart:39]: "Remember Me" is prefilling form fields.
"Keep Me Signed In" is the stay connected functionality you are referring to.. which is always on by default (for apps).

File: lib/auth/login/widgets/login_form.dart (hunk 3 above)
Comment (stephenrichter) [reply lib/auth/login/widgets/login_form.dart:78]: A couple of reasons I avoided it:
1. The UI is actually worse (spacing and layout issues)
2. It is tied to the init() state, same as email (which rely on past rememberMe settings)

File: lib/shared/utils/image_strings.dart (hunk 5 above)
Comment (stephenrichter) [reply lib/shared/utils/image_strings.dart:2]: Not sure yet, our backoffice is half down. Can wait for a future PR.

**Pull request merged**